tools/cache_creator/cache_creator.cpp

```diff
@@ -161,6 +162,82 @@ bool hexStringToUuid(llvm::StringRef hexStr, llvm::MutableArrayRef<uint8_t> outU
   return true;
 }
 
+static llvm::Error createBadPalMetadataError() {
+  return llvm::createStringError(llvm::object::object_error::invalid_file_type, "Unexpected PAL metadata format");
+}
+
+static llvm::Expected<llvm::msgpack::MapDocNode> get128BitCacheHashNode(llvm::msgpack::Document &document) {
+  llvm::msgpack::DocNode node = document.getRoot();
+  if (!node.isMap())
+    return createBadPalMetadataError();
+  node = node.getMap(false)["amdpal.pipelines"];
+  if (!node.isArray())
+    return createBadPalMetadataError();
+  node = node.getArray(false)[0];
+  if (!node.isMap())
+    return createBadPalMetadataError();
+  node = node.getMap(false)[".xgl_cache_info"];
```

Comment:
Sorry, I made a mistake in the last review.  A better name for this function is `getXglCacheInfoNode` because that is the node you are getting.